libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -6,89 +6,107 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Cognitive.LUIS.Models;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.Models;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
 using Newtonsoft.Json.Linq;
 
-namespace Microsoft.Bot.Builder.Ai.LUIS
+namespace Microsoft.Bot.Builder.Ai.Luis
 {
     /// <inheritdoc />
     /// <summary>
     /// A LUIS based implementation of IRecognizer.
     /// </summary>
     public class LuisRecognizer : IRecognizer
-    {
-        private const string MetadataKey = "$instance";
-        private readonly LuisService _luisService;
-        private readonly ILuisOptions _luisOptions;
-        private readonly ILuisRecognizerOptions _luisRecognizerOptions;
-
-        /// <summary>
+    {
+        /// <summary>
+        /// The value type for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceType = "https://www.luis.ai/schemas/trace";
+
+        /// <summary>
+        /// The context label for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceLabel = "Luis Trace";
+
+        private const string MetadataKey = "$instance";
+        private readonly LuisRuntimeAPI runtime;
+        private readonly LuisApplication application;
+        private readonly LuisPredictionOptions options;
+        private readonly bool includeApiResults;
+
+        /// <summary>
         /// Initializes a new instance of the <see cref="LuisRecognizer"/> class.
         /// </summary>
-        /// <param name="luisModel">The LUIS model to use to recognize text.</param>
-        /// <param name="luisRecognizerOptions">The LUIS recognizer options to use.</param>
-        /// <param name="options">The LUIS request options to use.</param>
-        public LuisRecognizer(ILuisModel luisModel, ILuisRecognizerOptions luisRecognizerOptions = null, ILuisOptions options = null)
-        {
-            _luisService = new LuisService(luisModel);
-            _luisOptions = options ?? new LuisRequest();
-            _luisRecognizerOptions = luisRecognizerOptions ?? new LuisRecognizerOptions { Verbose = true };
-        }
-
-        /// <inheritdoc />
-        public async Task<RecognizerResult> RecognizeAsync(string utterance, CancellationToken ct)
+        /// <param name="application">The LUIS application to use to recognize text.</param>
+        /// <param name="predictionOptions">The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">TRUE to include raw LUIS API response.</param>
+        public LuisRecognizer(LuisApplication application, LuisPredictionOptions predictionOptions = null, bool includeApiResults = false)
         {
```

Comment:
test & throw on nulls. TEsts to cover.

Reply:
Seems like there is some coverage here..
           Assert.ThrowsException<ArgumentException>(() => new LuisApplication(null, Guid.Empty.ToString(), "westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(string.Empty, Guid.Empty.ToString(), "Westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication("0000", Guid.Empty.ToString(), "Westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(Guid.Empty.ToString(), null, "Westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(Guid.Empty.ToString(), string.Empty, "Westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(Guid.Empty.ToString(), "0000", "Westus"));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(Guid.Empty.ToString(), Guid.Empty.ToString(), null));
            Assert.ThrowsException<ArgumentException>(() => new LuisApplication(Guid.Empty.ToString(), Guid.Empty.ToString(), "westus55"));